### DISCUSSION TITLE
'Backup' seems to contain data but does not show it

### DISCUSSION THREAD

**Discussion by pvswie** [psy0rz/zfs_autobackup#224] [General]
I finally found some time to continue on implementing ZFS snapshotting and a backup.
Setup: 
- VM with XigmaNAS (FreeBSD 13.2)
- 2 ZFS disks mounted as /mnt/pool/dataset and /mnt/pool-backup/dataset-backup

Command to backup pool to pool-backup:
 /usr/local/bin/zfs-autobackup -v offsite1 pool-backup
    which results in 'All operations completed successfully'

Looking at the target however no files are visible. 
ls -al /mnt/pool-backup/dataset-backup
total 1
drwxrwxrwx  3 root  wheel  2 Oct  5 08:50 .
drwxrwxrwx  3 root  wheel  3 Oct  5 08:50 ..
dr-xr-xr-x+ 3 root  wheel  3 Oct  5 08:50 .zfs

The .zfs folder also does not contain any files

What am I doing wrong?

===============================

Some more details
zfs get -t filesystem,volume autobackup:offsite1
NAME                                     PROPERTY             VALUE                SOURCE
pool                                     autobackup:offsite1  true                 local
pool-backup                              autobackup:offsite1  -                    -
pool-backup/dataset-backup               autobackup:offsite1  -                    -
pool-backup/dataset-backup/pool          autobackup:offsite1  true                 received
pool-backup/dataset-backup/pool/dataset  autobackup:offsite1  true                 inherited from pool-backup/dataset-backup/pool
pool/dataset                             autobackup:offsite1  true                 inherited from pool

zpool list -v
NAME          SIZE  ALLOC   FREE  CKPOINT  EXPANDSZ   FRAG    CAP  DEDUP    HEALTH  ALTROOT
pool         4.50G  11.1M  4.49G        -         -     0%     0%  1.00x    ONLINE  -
  ada1          5G  11.1M  4.49G        -         -     0%  0.24%      -    ONLINE
pool-backup  4.50G  22.1M  4.48G        -         -     0%     0%  1.00x    ONLINE  -
  ada2          5G  22.1M  4.48G        -         -     0%  0.47%      -    ONLINE

zfs --version
zfs-2.1.9-FreeBSD_g92e0d9d18
zfs-kmod-2.1.9-FreeBSD_g92e0d9d18

/usr/local/bin/zfs-autobackup --version
zfs-autobackup v3.2 - (c)2022 E.H.Eefting (edwin@datux.nl)

**Comment by psy0rz** [comment]
you have to mount the datasets, with zfs mount -a

this will be fixed in the next version. see #112

**Reply by pvswie** [reply]
Sorry I did not include that detail:
- I rebooted
- I ran zfs mount -a (as root) (before and after the reboot)
Neither had any effect. 

Any other ideas?
Peter

**Comment by psy0rz** [comment]
Well if you `zfs list -r -t snapshot`, and its there, you data should be there 100% as well.

Try explicitly mounting it with zfs mount.
